src/wallet/rpc/wallet.cpp

```diff
@@ -701,6 +701,59 @@ RPCHelpMan simulaterawtransaction()
     };
 }
 
+static RPCHelpMan migratewallet()
+{
+    return RPCHelpMan{"migratewallet",
+        "\nEXPERIMENTAL warning: This call may not work as expected and may be changed in future releases\n"
+        "\nMigrate the wallet to a descriptor wallet.\n"
+        "A new wallet backup will need to be made.\n"
+        "\nThe migration process will create a backup of the wallet before migrating. This backup\n"
+        "file will be named <wallet name>-<timestamp>.legacy.bak and can be found in the directory\n"
+        "for this wallet. In the event of an incorrect migration, the backup can be restored using restorewallet." +
+        HELP_REQUIRING_PASSPHRASE,
+        {},
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                {RPCResult::Type::STR, "wallet_name", "The name of the primary migrated wallet"},
+                {RPCResult::Type::STR, "watchonly_name", /*optional=*/true, "The name of the migrated wallet containing the watchonly scripts"},
+                {RPCResult::Type::STR, "solvables_name", /*optional=*/true, "The name of th migrated wallet containing solvable but not watched scripts"},
+                {RPCResult::Type::STR, "backup_path", "The location of the backup of the original wallet"},
+            }
+        },
+        RPCExamples{
+            HelpExampleCli("migratewallet", "")
+            + HelpExampleRpc("migratewallet", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
+    if (!wallet) return NullUniValue;
```

Comment:
nit: For new code, can use correct spacing?

```suggestion
        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
        {
            std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
            if (!wallet) return NullUniValue;
```

Reply:
Done